adhocteam/Head-Start-TTADP
# Add file upload api

**Description of change**
This PR adds the a API endpoint POST /files. This is only for the api, and does not include the front end.

**How to test**
1. Update .env file with new variables: 
    1. `AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY`
    2. `AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE`
    3. `MINIO_ROOT_PASSWORD=$AWS_SECRET_ACCESS_KEY`
    4. `MINIO_ROOT_USER=$AWS_ACCESS_KEY_ID`
    5. `S3_BUCKET=ttadp-test`
    6. `LOCAL_DEV=true`
    7. If you aren't using docker, `S3_ENDPOINT=http://localhost:9000'
2. Start app using `yard docker:start`
2. Go to localhost:9000 to access the Minio web console (use AWS_SECRET_ACCESS_KEY for password and AWS_ACCESS_KEY_ID for username).
3. create a bucket named `ttadp-test` (I would like to automate this step in the future)
4. save a Activity Report using the UI. Note the 
4. Open a terminal and type `curl localhost:3000/api/files -F File=@TESTFILE.txt -F reportID=1 -F attachmentType=ATTACHMENT` where TESTFILE.txt is a file in the directory you run the curl command from
5. Output should be similar to  
```
{
"id":5
}
```

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/150
  * Part of https://github.com/HHS/Head-Start-TTADP/issues/150. Needs frontend connected to api still.
* https://github.com/HHS/Head-Start-TTADP/issues/130
  * closes https://github.com/HHS/Head-Start-TTADP/issues/130 as it checks that versioning is enabled with every file upload.

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
